pkg/util/workerqueue/workerqueue.go

```diff
@@ -37,6 +37,29 @@ const (
 	workFx = time.Second
 )
 
+// DebugError is a marker type for errors that that should only be logged at a Debug level.
+// Useful if you want a Handler to be retried, but not logged at an Error level.
+type DebugError struct {
+	err error
+}
+
+// NewDebugError returns a DebugError wrapper around an error.
+func NewDebugError(err error) error {
+	return &DebugError{err: err}
+}
+
+// Error returns the error string
+func (l *DebugError) Error() string {
+	return l.err.Error()
```

Comment:
Do you need to add `if l.err = nil return nil` before this line?

Reply:
Doesn't hurt! Don't want the panics!